src/vs/platform/backup/electron-main/backupMainService.ts

```diff
@@ -132,9 +134,26 @@ export class BackupMainService implements IBackupMainService {
 
 		backups.folderWorkspaces.forEach(workspacePath => {
 			const backupPath = path.join(this.backupHome, this.getWorkspaceHash(workspacePath));
-			if (!this.hasBackupsSync(backupPath)) {
+			const hasBackups = this.hasBackupsSync(backupPath);
+			const missingWorkspace = hasBackups && !fs.existsSync(workspacePath);
+
+			// If the folder has no backups, make sure to delete it
+			// If the folder has backups, but the target workspace is missing, convert backups to empty ones
+			if (!hasBackups || missingWorkspace) {
 				const backupWorkspace = this.sanitizePath(workspacePath);
 				staleBackupWorkspaces.push({ workspaceIdentifier: Uri.file(backupWorkspace).fsPath, backupPath, isEmptyWorkspace: false });
+
+				if (missingWorkspace) {
+					const identifier = this.pushBackupPathsSync((Date.now() + Math.round(Math.random() * 1000)).toString(), true /* is empty workspace */);
```

Comment:
Should we add this random part to the regular folder generation in `registerWindowForBackups` if you're concerned about conflicts?